### ISSUE TITLE
Snarl traversal output incomplete

### ISSUE THREAD

**Issue by egoltsman** [vgteam/vg#3091]
Hi vg folks,
The ability to extract snarl traversals is a really nice feature as it lets you (potentially) access variants right in the graph data structure, which is much more portable and meaningful than going by allele info in the "deconstructed" vcf file.   However, the file I get with 'vg snarls -r [traversals_file] -e -a [graph]  `   is missing some important data.   Here's an example (text-ified from protoBuf with 'vg view -E [traversals_file] )
```
{"visit": [{"node_id": "8379"}, {"node_id": "12303"}, {"node_id": "11670"}, {"node_id": "12304"}, {"node_id": "8382"}]}
```
Looking at the vg.proto file, it seems like, when serialized to protoBuf,  the traversal has a 'name' attribute, and the visited nodes  have some orientation info (bool backward ).   But for whatever reason, they don't get extracted into the json record.  Without these the path seems ambiguous (it can't reveal an inversion, for example)    I suspect this is just a matter of handling the serialization differently, but I must admit the whole protobuf thing is pretty fuzzy to me.

Also, the -p/--pathnames option doesn't seem to work.  Nothing gets printed to the traversals file (if  -r [file] is included), and the SDOUT ends up having some formatting mismatch issue when I try to view it:

```
vg snarls -t 12 -T -p -a -e -r foo.traversals [graph] > foo.snarls.pb
vg view -R foo.snarls.pb
terminate called after throwing an instance of 'std::runtime_error'
  what():  [io::ProtobufIterator] tag "vg.SnarlTraversal" for Protobuf that should be "SNARL"
ERROR: Signal 6 occurred. VG has crashed. Run 'vg bugs --new' to report a bug.
Stack trace path: /tmp/vg_crash_eSDa7Q/stacktrace.txt
Please include the stack trace file in your bug report!
```

Thanks!

**Comment by jeizenga** [comment]
Hi Eugene,

You're getting into some code that we haven't used actively in quite a while, so some of it might have gone a bit stale. As for the "name" field, I don't think we ever fully utilized it. However, the sequence of Visits should be enough to unambiguously indicate which snarl a SnarlTraversal is referring to, since snarls are uniquely identified by their pair of boundary nodes. Also, the Protobuf format we use for serializing SnarlTraversals actually doesn't distinguish between the default value of "false" and having no value, so when the "backward" isn't included, that means the Visit is forward. 

As for the `-p` option, it looks to me like the help documentation says that it causes the SnarlTraversals to go to stdout, which is what that crash shows as well. VG should probably give a warning about this behavior when both `-p` and `-r` are provided. I'll confess that it's not totally clear to me what the "path name" terminology means in this context though. Maybe @glennhickey could comment?

Hope that helps!

**Comment by egoltsman** [comment]
Without the -r flag it rejects the -e flag, and that one is critical for me since I'm only interested in sample-coherent traversals, i.e. those that match a slice of a Path in the graph.   These are the "path names" that I assume the -p  option refers to.  But that's just my understanding.  
Thanks

**Comment by glennhickey** [comment]
The `-p` option stale and should be removed.  `-r -e` will report every path through the snarl.  As @jeizenga said `False` and `0`-values are not printed, so your example traversal is not ambiguous, it is all on the forward strand.

**Comment by egoltsman** [comment]
Ok, thanks.  So, in a traversal, the snarl name should always correspond to the start+end  node pair, and never the other way around, correct?
Now, in the snarls file, I notice (at least in this specific case)  the the 'backward' tag is allays present in the *parent*  snarl if there's nesting,  and never with the child snarl.  It's also always missing from  unnested standalole snarls.   Is this normal?  I guess I'm still struggling to understand how directionality should be interpreted here.  
Thank you!

**Comment by jeizenga** [comment]
I don't know of any consistent pattern of representing parent/child snarls with backward.

There's an important sense in which the designation of "backward" and "forward" to a snarl is kind of arbitrary in these graphs, so I would caution against reading too much into this field. An analogous point is that on a conventional linear reference, the designation of one strand as the "forward" strand is just an informatic convenience. The other strand could have just as easily been labeled the forward strand in the reference, but it's helpful to choose one so that we can all refer to the two strands in consistent terminology. It's the same in these graphs. One direction through the snarl is as good a choice for "forward" as the other, but we choose one to be canonically forward for convenience. The orientation of the snarl is only ever intrinsically important relative to some walk through the graph.

**Comment by glennhickey** [comment]
> So, in a traversal, the snarl name should always correspond to the start+end node pair, and never the other way around, correct?

Yes, the link between a traversal and a snarl is the traversal's endpoints, which are in the same order and orientation as the snarl's start and end.  

This orientation is itself arbitrary as Jordan says.  In particular the snarl `Start=(id:1, forward) End=(id:5, forward)`  is identical to  `Start=(id:5, backward) End=(id:1, backward)`.  But the orientation of the snarl will always be consistent with the orientation of the traversals.

**Comment by egoltsman** [comment]
That's what I figured, thanks for confirming.  Now, is it also possible to have a snarl with a forward start and a backward end  ( and vise versa) ?  In theory, I don't see why not, but can you please confirm?

**Comment by jeizenga** [comment]
Yes, that is definitely possible.

**Comment by egoltsman** [comment]
Hi Glen, Jordan,
Going back to the original topic, it would be really great to have the -p option revived, or have some other way of matching a snarl traversal with the underlying path(s).  If this can already be done, please advise! 
Thanks again!

**Comment by faithokamoto** [comment]
`-r -e` does what you want -- it limits snarl traversals to those which correspond to an embedded path. At least that's what the comments in the code say. I'm not sure what `-p` does (it has no tests at all) but if Glenn said it was stale in 2020 then it's probably stale *now* in 2026.

**Comment by faithokamoto** [comment]
`vg snarls --pathnames` removed
